src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -936,10 +960,54 @@ public void setHealth(double health) {
         }
     }
 
+    @Override
+    public void damage(double amount, Block block, DamageCause cause) {
+        if (!canTakeDamage() || !canTakeDamage(cause)) {
+            return;
+        } else {
+            noDamageTicks = maximumNoDamageTicks;
+        }
+
+        amount = getEffectiveDamage(amount);
+
+        // fire event
+        EntityDamageByBlockEvent event = EventFactory.getInstance().onEntityDamage(
+                new EntityDamageByBlockEvent(block, this, cause, amount));
+
+        if (event.isCancelled()) {
```

Comment:
Still some duplicate code here. Couldn't we pass in the `EntityDamageEvent` to a new function?

Reply:
Duplicated code is really short and it isn't worth to make a new function to allow both events, they are much different and their handling is chopped